async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -365,6 +366,60 @@ impl<'a> futures::Stream for Stream<'a> {
     }
 }
 
+pub struct StreamBuilder<'a> {
+    batch: usize,
+    max_bytes: usize,
+    hearbeat: Duration,
+    expires: usize,
+    consumer: &'a Consumer<Config>,
+}
+
+impl<'a> StreamBuilder<'a> {
+    pub fn new(consumer: &'a Consumer<Config>) -> Self {
+        StreamBuilder {
+            consumer,
+            batch: 200,
+            max_bytes: 0,
+            expires: 0,
+            hearbeat: Duration::default(),
+        }
+    }
+
+    pub fn max_bytes_per_batch(mut self, max_bytes: usize) -> Self {
+        self.max_bytes = max_bytes;
+        self
+    }
+
+    pub fn max_messages_per_batch(mut self, batch: usize) -> Self {
+        self.batch = batch;
+        self
+    }
+
+    pub fn hearbeat(mut self, hearbeat: Duration) -> Self {
+        self.hearbeat = hearbeat;
+        self
+    }
+
+    pub fn expires(mut self, expires: Duration) -> Self {
```

Comment:
We know there is a bug with `expires` in the server, but I still think we should expose that methods, as it's not client err. Also, not making it public forces us to remove (ignored) timeout test.